source/slang/source-loc.h

```diff
@@ -45,12 +57,20 @@ class SourceLoc
 
 inline SourceLoc operator+(SourceLoc loc, Int offset)
 {
-    return SourceLoc::fromRaw(loc.getRaw() + UInt(offset));
+    return SourceLoc::fromRaw(SourceLoc::RawValue(Int(loc.getRaw()) + offset));
 }
 
 // A range of locations in the input source
 struct SourceRange
 {
+        /// True if the loc is in the range. Range is inclusive on begin to end.
+    bool contains(SourceLoc loc) const { const auto rawLoc = loc.getRaw(); return rawLoc >= begin.getRaw() && rawLoc <= end.getRaw(); }
```

Comment:
`contains()` might actually be a tough query, which could have to go back to the `SourceManager`. The question is: should a location in one `#include` of a file be contained in a range that covers the "same" bytes in a different `#include` of the file?

The current definition is fine, but exposing it as a query directly on `SourceRange` gives the implication that this is a generally useful query, where it isn't clear what users can/should do with it in most cases.